### PR TITLE
fix RSS Feed - first load not working

### LINKED ISSUE #25211: RSS Feed - first load not working
### Preconditions (*)

1. Magento 2.3.x & 2.3-develop
2. PHP 7.2.21 (i believe any other version too).

### Steps to reproduce (*)

1. Enable standard RSS Feed module.
2. Enable RSS Feed for New Products and Special Products.

3. Proceed to domain.tld/rss
image
4. Choose any channel.(New product in my case)
5. We get an exception
image
6. Refresh of site will make RSS Feed work. The point is that it's not working every time after compile.
image

### Expected result (*)

1. Module should output XML file

### Actual result (*)

> 1 exception(s):
Exception #0 (Zend\Feed\Writer\Exception\InvalidArgumentException): Invalid parameter: parameter must be a non-empty string

### What i found (*)
I debugged it and in Zend\Feed\Writer\AbstractFeed in method setTitle there is condition:
`        if ((empty($title) && ! is_numeric($title)) || ! is_string($title)) {
            throw new Exception\InvalidArgumentException('Invalid parameter: parameter must be a non-empty string');
        }`

$title is an instance of 'Magento\Framework\Phrase' that is why it's throwing an exception.
When i changed the condition to:

`if ((empty($title) && ! is_numeric($title)) || (! is_string($title) && strpos(get_class($title), 'Magento\Framework\Phrase') === false)) {`

I tried this way just temporary for tests but it will require to modify data before it will reach Zend's library method.

### PR DESCRIPTION
**Pull request by engcom-Charlie**
### Description (*)

Fixed and covered by tests

### Fixed Issues (if relevant)

1. Fixes magento/magento2#25211

### Manual testing scenarios (*)

1. see #25211

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by engcom-Charlie** [comment]
@magento run all tests

File: app/code/Magento/Rss/Model/Rss.php [hunk 1]
```diff
@@ -59,54 +62,56 @@ public function __construct(
...
-                $this->dataProvider->getCacheKey(),
-                ['rss'],
-                $this->dataProvider->getCacheLifetime()
-            );
+        if ($cacheKey && $cacheLifeTime) {
+            $this->cache->save($serializedData, $cacheKey, ['rss'], $cacheLifeTime);
         }
 
-        return $data;
+        return $this->serializer->unserialize($serializedData);
```
Comment (ihor-sviziev) [inline app/code/Magento/Rss/Model/Rss.php:89]: why do we need to serialize it and unserialize again?

Comment (engcom-Charlie) [reply app/code/Magento/Rss/Model/Rss.php:89]: To be sure that it returns an array that will not consist Phrase object. And also it is one place to fix NewProducts, Wishlist, Product Special, etc.

Comment (ihor-sviziev) [reply app/code/Magento/Rss/Model/Rss.php:89]: Ok, it’s not obvious, in this case would be good to add comment

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-7999](https://jira.corp.magento.com/browse/ENGCOM-7999) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Manual testing scenario:**
1. Enable standard RSS Feed module;
2. Enable RSS Feed for New Products and Special Products;


3. Proceed to domain.tld/rss

image
4. Choose any channel. (New product in my case)

**Before:** :heavy_multiplication_x: We get an exception

Screenshot from 2020-08-12 09-43-49

**After:** :heavy_check_mark: The module outputs an XML file

Screenshot from 2020-08-12 10-20-10

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Pull request merged**